File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOErrorRecordModel.kt
//
//  IDOErrorRecordModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get Error Records event number

open class IDOErrorRecordModel(type: Int, resetFlag: Int, hwError: Int) : IDOBaseModel {

    /// 0 Query
    /// 1 Clear Log
    @SerializedName("type")
    var type: Int = type

    /// 0 Normal
    /// 1 Hard Faul
    /// 2 Watchdog service
    /// 3 Assertion reset
    /// 4 Power-off service
    /// 5 Other exceptions
    @SerializedName("reset_flag")
    var resetFlag: Int = resetFlag

    /// Hardware error code
    /// 0 Normal
    /// 1 Accelerometer error
    /// 2 Heart rate error
    /// 3 TP error
    /// 4 Flash error
    @SerializedName("hw_error")
    var hwError: Int = hwError


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOErrorRecordModel.swift
//
//  IDOErrorRecordModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get Error Records event number
@objcMembers
public class IDOErrorRecordModel: NSObject, IDOBaseModel {
    /// 0 Query
    /// 1 Clear Log
    public var type: Int
    /// 0 Normal
    /// 1 Hard Faul
    /// 2 Watchdog service
    /// 3 Assertion reset
    /// 4 Power-off service
    /// 5 Other exceptions
    public var resetFlag: Int
    /// Hardware error code
    /// 0 Normal
    /// 1 Accelerometer error
    /// 2 Heart rate error
    /// 3 TP error
    /// 4 Flash error
    public var hwError: Int
    
    enum CodingKeys: String, CodingKey {
        case type = "type"
        case resetFlag = "reset_flag"
        case hwError = "hw_error"
    }
    
    public init(type: Int,resetFlag: Int,hwError: Int) {
        self.type = type
        self.resetFlag = resetFlag
        self.hwError = hwError
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

